unit/getinfo_test.c

```diff
@@ -143,20 +143,43 @@ fail:										\
  * Tests:
  */
 
-getinfo_test(1, "Test with no node, service or flags",
```

Comment:
I've added that test for no hints case as well. Also added a comment and warning message that some of the tests are not  run for a specific provider.